internal/controllers/clusterrequest_controller.go

```diff
@@ -1723,10 +1720,10 @@ func (t *clusterRequestReconcilerTask) updateClusterInstance(ctx context.Context
 // waitForHardwareData waits for the NodePool to be provisioned and update BMC details
 // and bootMacAddress in ClusterInstance.
 func (t *clusterRequestReconcilerTask) waitForHardwareData(ctx context.Context,
-	clusterInstance *siteconfig.ClusterInstance, nodePool *hwv1alpha1.NodePool) bool {
+	clusterInstance *siteconfig.ClusterInstance, nodePool *hwv1alpha1.NodePool) (bool, bool, error) {
 
-	provisioned := t.checkNodePoolProvisionStatus(ctx, nodePool)
-	if provisioned {
+	provisioned, timedOutOrFailed, err := t.checkNodePoolProvisionStatus(ctx, nodePool)
```

Comment:
Should we log the err here since the inner error logs are removed?

Reply:
We have this  error log "Failed to update the NodePool status for ClusterRequest" in checkNodePoolProvisionStatus().

Reply:
Okay, I was considering also logging the error returned from utils.DoesK8SResourceExist, but since we requeue on error, we should get the trackback in log for any temporary API issues. 
However, I think err var is missing in the log "Failed to update the NodePool status for ClusterRequest"

Reply:
The err var is added to the log.